### ISSUE TITLE
CMap high bit surrogate range checks should probably be inclusive 

### ISSUE THREAD

**Issue by alishersadikov** [yob/pdf-reader#499]
First of all, thank you for this great library. 

We have recently run into an interesting edge case while parsing a PDF file with a CMAP specified for a font, it looked like this: 

```
#<PDF::Reader::CMap:0x00007fba3692da40 @map={2755=>[55296, 56578]}>
```

It was used to display `𐄂`  in the PDF, represented in the CMAP as the bytes [55296, 56578] or "\xD8\x00\xDD\x02". We also noticed this range check in `PDF::Reader::Cmap`: https://github.com/yob/pdf-reader/blob/main/lib/pdf/reader/cmap.rb#L120-L122, which would exclude 55296 when it is equal to 0xD800: 

```
unpacked_string.first.to_i == 0xD800
```

Therefore, we think those high bit surrogate range checks should be inclusive. I can open a PR to address this issue if it makes sense.

**Comment by yob** [comment]
> Therefore, we think those high bit surrogate range checks should be inclusive. I can open a PR to address this issue if it makes sense.

I'm not an expert at UTF-16 and surrogate pairs, but at face value this seems to be a reasonable plan. I'll happily accept a PR - can you include specs in `spec/cmap_spec.rb`? Maybe copying `spec/data/cmap_with_surrogate_pairs.txt` to a new fixture that includes the boundary conditions (`0xD800` and `0xD8FF`) would make sense.

**Comment by yob** [comment]
I came across a PDF with this issue earlier this week and was able to create a simple test case: #529. The fix has been release in v2.12.0
